File: backend/src/main/java/com/companyshield/demo/service/UserService.java
package com.companyshield.demo.service;


import com.companyshield.demo.domain.dto.UserDto;
import com.companyshield.demo.domain.transactional.CreateUserRequest;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public interface UserService {
    boolean createUser(CreateUserRequest createUserRequest);
    Optional<UserDto> getUserById(String id);
    List<UserDto> getAllUsers();
}

File: frontend/src/service/user_service.ts
import { GetAllUsersResponse } from "../domain/transactional/GetAllUsersResponse";
import { GetUserResponse } from "../domain/transactional/GetUserResponse";

export interface UserService {
    getAllUsers(): Promise<GetAllUsersResponse | Error>;
    getCurrentUser(): Promise<GetUserResponse | Error>;
    createUser(username: string, password: string): Promise<boolean>;
}